File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let left = 0;
    let pointer = 0;

    while (pointer < nums.length){
        if(nums[pointer] != 0){
            if (pointer != left){
                let temp = nums[left]
                nums[left] = nums[pointer]
                nums[pointer] = temp
            }
            left++
        }
        pointer++
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
      // Start with two pointers 
      int left = 0;
      int pointer = 0;
      if(nums.length == 1) return;
      // If you encounter non-0 value, move it to the front  
      while(pointer <= nums.length- 1){
      if (nums[pointer] != 0){
        int copy = nums[left];
        nums[left] = nums[pointer];
        nums[pointer] = copy;
        left++;
      }
      pointer++;

      }
    }
}